File: 9041 - Software Instruction/lab07/total_words.py
#!/usr/bin/python

import re,sys

count = 0
for line in sys.stdin:
	words = re.split(r"[^a-zA-Z]", line)
	for word in words:
		pattern = re.compile("^$")
		if not pattern.match(word):
				count += 1



print str(count) + " words"

File: 9041 - Software Instruction/lab07/total_words.pl
#!/usr/bin/perl -w

@input = <STDIN>;
$words = 0;

sub removeEmptyStr
{
	$count = 0;
	foreach $str (@_)
	{
		chomp $str;
		if ($str eq "" or $str =~ /^\s+$/)
		{
			$count = $count + 0;	
			next;
		}
		$count++;
	}
	return $count;	
}

foreach $line (@input)
{
	chomp $line;
	$line =~ s/\s+/ /;
	if ($line eq "" or $line=~/^ *$/ or $line =~ /^\s*$/) {
		next;
	}
	@newLine = split(/[^a-zA-Z]+/, $line);	
	$words += removeEmptyStr(@newLine);
}

print "$words words\n";

